web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte

```diff
@@ -115,13 +115,18 @@
   }
 
   const onSaveAndRefreshClick = async (tableName: string) => {
+    // no-op if the clientYAML is still null
+    if ($sourceStore.clientYAML === null) return;
     overlay.set({ title: `Importing ${tableName}.yaml` });
     await saveAndRefresh(tableName, $sourceStore.clientYAML);
     checkSourceImported(queryClient, sourceName, filePath);
     overlay.set(null);
   };
 
   const onRefreshClick = async (tableName: string) => {
+    // no-op if connector is undefined
+    if (connector === undefined) return;
```

Comment:
we can disable the button, but we can't take out this check because of the way the component is set up. In practice, I think this code path will never actually be hit anyway. This whole thing is ugly anyway (see eg the try/catch with a no-op directly below) and could use a refactor, but I'm gonna make the call that that's out of scope for this PR